src/transaction.js

```diff
@@ -17,12 +17,19 @@ export class Transaction {
     };
   }
 
+  formatTransaction() {
+    if (typeof this.credit == "number") {
+      return `${this._formatDate()}|| ${this.credit.toFixed(
+        2
+      )} || || ${this.balance.toFixed(2)}`;
+    } else if (typeof this.debit === "number") {
+      return `${this._formatDate()}|| || ${this.debit.toFixed(
+        2
+      )} || ${this.balance.toFixed(2)}`;
+    }
+  }
```

Comment:
it feels like this method is doing a lot, do you think you could break it down any further?

Reply:
It does feel a little jumbled here, maybe worth just reformatting it just for sake of reliability  :)